backend/src/apiserver/resource/resource_manager.go

```diff
@@ -410,6 +461,23 @@ func (r *ResourceManager) CreateRun(ctx context.Context, apiRun *api.Run) (*mode
 	return r.runStore.CreateRun(runDetail)
 }
 
+type owner struct {
+	name string
+	id   string
+}
+
+func getOwner(resourceRef []*api.ResourceReference) owner {
+	for _, ref := range resourceRef {
+		if ref.Relationship == api.Relationship_OWNER {
```

Comment:
nit: this can also mean an owner in KFP resource relationship. I think for adding Kubernetes owners, we should only add it for "runs owned by jobs".